src/CsFixerConfig.php

```diff
@@ -117,16 +117,18 @@ public static function create(): ConfigInterface
     /**
      * @param array<string, mixed> $rules
      */
-    public function addRules(array $rules): ConfigInterface
+    public function addRules(array $rules): self
     {
         $rules = array_replace_recursive($this->getRules(), $rules);
-        return $this->setRules($rules);
+        $this->setRules($rules);
+
+        return $this;
     }
 
     public function setHeader(
```

Comment:
Here, we can add this annotation to make it visible that this method uses a fluent interface:

```php
@return $this
```

Reply:
Thanks, will check...

Reply:
Well, does not work for stan:

```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/CsFixerConfig.php                                                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
  124    Method TYPO3\CodingStandards\CsFixerConfig::addRules() should return $this(TYPO3\CodingStandards\CsFixerConfig) but returns PhpCsFixer\ConfigInterface.   
  144    Method TYPO3\CodingStandards\CsFixerConfig::setHeader() should return $this(TYPO3\CodingStandards\CsFixerConfig) but returns PhpCsFixer\ConfigInterface.  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
 ```

Reply:
Ah, this seems to be a problem with the type declarations of `vendor/friendsofphp/php-cs-fixer/src/Config.php`. So we cannot do this.